rosdep/python.yaml

```diff
@@ -3119,6 +3119,15 @@ python-setuptools:
     xenial_python3: [python3-setuptools]
     yakkety: [python-setuptools]
     zesty: [python-setuptools]
+python-sh:
+  debian: [python-sh]
+  fedora: [python-sh]
+  gentoo: [dev-python/sh]
+  ubuntu:
+    '*': [python-sh]
```

Comment:
This wildcard will also match Ubuntu distros earlier than Trusty. And for those the resulting mapping will be invalid. Therefore imo this should enumerate the older Ubuntu distros explicitly with a mapping to `None`.

Reply:
That's what we've been doing for new keys for a while: Keep rule simple and not adding explicitly null entries for every distro that was already EOL when the key was added.
Did that change ?

Reply:
When not using a wildcard the missing entry just means rosdep can't resolve the dependency on that platform. That was just fine. But when using `'*'` the impact is very different because not existing mappings will fall back to the wildcard which is wrong in that case. Therefore when using the wildcard "unsupported" distros should be explicitly listed as "not supported" to avoid that wrong mapping.

Reply:
Isnt that the exact same thing as adding `ubuntu: [foo]` ? it will be resolved for all ubuntu platforms even the EOL ones. And that;s what we've been doing for a while as long as it was defined for all non EOL distros.

I dont think expecting people to list explicitly every EOL distros when adding a new key is desired.

Reply:
It has been our policy that if a key is added it's ok to use the generic matches all if all non-EOL distros support that specific key. This will cause EOL distros to incorrectly resolve the keys on the older distros. However that will not break any existing users, because the key never existed. So by definition it's not being used on the EOL distros, thus we will not break anyone on the EOL distros. (except for a name collision with a private package or rosdep keys but that's on them and easily resolvable.)

Our goal for keeping the EOL'd keys around is to not break people using them. This continues to not break existing users. It's just something that if you're actively developing or backporting to an EOL platform you may run into resolutions that are not expected, but you're going outside of our supported envelope and some work may be required.